pkg/storage/operational_solutionStore.go

```diff
@@ -72,6 +77,27 @@ func (s *Store) OperationalSolutionGetByID(_ *zap.Logger, id uuid.UUID) (*models
 	return &solution, err
 }
 
+// OperationalSolutionGetByIDLOADER returns an operational solution by ID using a DataLoader
+func (s *Store) OperationalSolutionGetByIDLOADER(
+	_ *zap.Logger,
+	paramTableJSON string,
+) ([]*models.OperationalSolution, error) {
+	arg := map[string]interface{}{
+		"paramTableJSON": paramTableJSON,
+	}
+
+	opSols, err := sqlutils.SelectProcedure[models.OperationalSolution](
+		s.db,
+		sqlqueries.OperationalSolution.GetByIDLOADER,
+		arg,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("issue selecting operational solution by id with the data loader, %w", err)
```

Comment:
This might be a good case for using the *zap.Logger passed in here -- perhaps something like this! (Though this commit won't work directly)

```suggestion
		errorMessage := "error selecting operational solution by id"
		logger.Error(errorMessage, zap.Error(err))
		return nil, errors.Wrap(err, errorMessage)
```

This allows us to see the trace of the error through the logs, but also retains the error messages context through the call chain!